### PR TITLE
Elasticsearch geo location type addition

### PR DESCRIPTION
**Pull request by PBMikeW**
Add switch statement and dynamic template for GeoPoint -> geo_point. Uses search pattern of *.Location as GeoPointFieldIndexHandler forces an index as "Location"

Mapping:
`      "dynamic_templates": [
        {
          "*.Location": {
            "path_match": "*.Location",
            "match_mapping_type": "object",
            "mapping": {
              "type": "geo_point"
            }
          }
        }
      ],`

Query:
`{
  "query": {
    "bool": {
      "must": [
        {
          "geo_bounding_box": {
            "ContentTypeName.Location": {
              "top_left": {
                "lat": -34.0,
                "lon": 166.0
              },
              "bottom_right": {
                "lat": -40.0,
                "lon": 179.0
              }
            }
          }
        }
      ]
    }
  }
}`

### CODE REVIEW

**Comment by PBMikeW** [comment]
@Skrypt

**Comment by Skrypt** [comment]
Will review

File: src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexManager.cs [hunk 1]
```diff
@@ -571,6 +583,12 @@ private static Dictionary<string, object> CreateElasticDocument(DocumentIndex do
                         }
                     }
                     break;
+                case DocumentIndex.Types.GeoPoint:
+                    if (entry.Value is DocumentIndex.GeoPoint point)
+                    {
+                        AddValue(entries, entry.Name, new GeoLocation((double)point.Latitude, (double)point.Longitude));
```
Comment (Skrypt) [inline src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexManager.cs:589]: Use long for accuracy?

**Review by Skrypt — APPROVED** [review]

File: src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexManager.cs [hunk 2]
```diff
@@ -571,6 +583,12 @@ private static Dictionary<string, object> CreateElasticDocument(DocumentIndex do
...
+                    }
+                    break;
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexManager.cs:591]:
```suggestion

                    break;
```

File: src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexManager.cs [hunk 3]
```diff
@@ -191,6 +191,18 @@ await _elasticClient.MapAsync<string>(p => p
...
             );
+        // DynamicTemplates mapping for Geo fields, the GeoPointFieldIndexHandler adds a Location index by default.
+        await _elasticClient.MapAsync<object>(p => p
+            .Index(fullIndexName)
+            .DynamicTemplates(d => d
+                .DynamicTemplate("*.Location", dyn => dyn
+                    .MatchMappingType("object")
+                    .PathMatch("*" + ".Location")
+                    .Mapping(m => m
+                        .GeoPoint(g => g))
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexManager.cs:202]:
```suggestion
                    .Mapping(m => m.GeoPoint(g => g))
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Review by hishamco — APPROVED** [review]

**Pull request merged**